azure-functions/acbs-function/mappings/facility/facility-covenant.js

```diff
@@ -10,6 +10,21 @@
 const helpers = require('./helpers');
 const CONSTANTS = require('../../constants');
 
+/**
+ * Maps the facility covenant record for a given deal and facility.
+ *
+ * This function performs the following operations:
+ * 1. Extracts the guarantee expiry date and effective date from the facility's guarantee dates.
+ * 2. Retrieves the currency of the facility, defaulting to a constant if not provided.
+ * 3. Calculates the maximum liability using a helper function.
+ * 4. Constructs and returns the facility covenant record object.
+ *
+ * @param {Object} deal - The deal object containing deal details.
+ * @param {Object} facility - The facility object containing facility details.
+ * @param {string} covenantType - The type of the covenant.
+ * @returns {Object} - The mapped facility covenant record, including covenant type, maximum liability, currency, guarantee expiry date, and effective date.
+ * @throws {Error} - Logs the error and returns an empty object if any error occurs during the mapping process.
```

Comment:
I wont comment everywhere regarding the `@throws` -- but I do think we shouldn't say things throw an error when we're using other error handling (the return of an empty object) to handle errors as currently this would read as an Error object being thrown